### ISSUE TITLE
Double underscore module variables are ignored?

### ISSUE THREAD

**Issue by Andrej730** [microsoft/pyright#8131]
See example below - importing from a module a variable that has two leading underscores Pylance shows an error:

```
"__test" is unknown import symbol Pylance[reportAttributeAccessIssue]
(import) __test: Unknown
```

Though variable is there and it will be imported fine and can be accessed without a mangling (in python module variable starting with underscore are not imported with a wildcard import but allowed otherwise). Even specifying variable in `__all__` doesn't help.

![image](https://github.com/microsoft/pylance-release/assets/9417531/846821e3-35cb-4066-bc4d-fcecc6323027)

**Comment by erictraut** [comment]
I agree that double underscore symbols that are part of a module namespace should not be ignored unless they are part of a type stub or a "py.typed" package.

Double underscore symbols in a class namespace do need to be treated specially, however, because they are mangled.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.368.

**Comment by Andrej730** [comment]
Tested with 1.1.368 - it works, thank you! 

But I think there is one more caveat to this - when you use any name with double underscore in the class body or in the class methods, it gets mangled. So, it shouldn't recognize `__test` inside class body and it also shouldn't recognize other classes attributes cause they get mangled too (e.g. `B.__test`). It would be nice if it could clearify in the tooltip that it doesn't recognize existing symbols because of the inside-class mangling as it probably may be confusing for some users.

Btw, is it intended that in b.py with `pyright: reportUnusedVariable=true` `_test` and `__test` are recognized as unused variables but `test1` isn't?

![image](https://github.com/microsoft/pyright/assets/9417531/2901fcfd-93cc-43b6-b1f4-22adc8d7424c)


Snippets:
```python
from b import _test, __test

class B:
    __test = 99


class A:
    __a = 25
    # NameError: name '_A__test' is not defined. Did you mean: '__test'?
    print(__test)

    def __init__(self):
        # NameError: name '_A__test' is not defined. Did you mean: '__test'?
        print(__test)
        # AttributeError: type object 'B' has no attribute '_A__test'. Did you mean: '_B__test'?
        print(B.__test)

a = A()
```

```python
# pyright: reportUnusedVariable=true

test1 = 35
_test = 25
__test = 45
```

**Comment by erictraut** [comment]
Yes, this is a subtle side effect of Python's compiler. I'm not aware of any static type checkers or linters that model this behavior. It's a feature that could be added if there's sufficient demand, but I don't think this is a common use case.

> is it intended that in b.py with pyright: reportUnusedVariable=true _test and __test are recognized as unused variables but test1 isn't?

Yes. The symbols with underscores are assumed to be private to the module — not exported to other modules.
